### PR TITLE
Make ShipwrightBuild cluster-scoped

### PR DESCRIPTION
**Pull request by adambkaplan**
# Changes

- Make `ShipwrightBuild` a cluster-scoped CRD
- Add verification checks to ensure artifacts are correctly generated.

/kind feature

# Submitter Checklist

- [x] Includes tests if functionality changed/was added
- [x] Includes docs if changes are user-facing
- [x] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)  
- [x] Release notes block has been filled in, or marked NONE

See [the contributor guide](https://github.com/shipwright-io/build/blob/master/CONTRIBUTING.md)
for details on coding conventions, github and prow interactions, and the code review process.

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Review by adambkaplan — COMMENTED** [review]
/assign @otaviof 

/cc @coreydaley @gabemontero

**Comment by adambkaplan** [comment]
/approve

Self-approving

File: bundle/manifests/operator.shipwright.io_shipwrightbuilds.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ spec:
     listKind: ShipwrightBuildList
     plural: shipwrightbuilds
     singular: shipwrightbuild
-  scope: Namespaced
+  scope: Cluster
```
Comment (gabemontero) [inline bundle/manifests/operator.shipwright.io_shipwrightbuilds.yaml:15]: IIRC openshift-pipelines exposes the option when you install the the operator from the console of being cluster wide or per namespace.

If I'm correct, I wonder if we should follow that pattern.

And if so, figure out exactly how they do it.

I'm fine with putting a TODO comment and opening a follow up feature if you like @adambkaplan

Comment (adambkaplan) [reply bundle/manifests/operator.shipwright.io_shipwrightbuilds.yaml:15]: @gabemontero OLM operators initially allow operators to be namespaced, in which case the operator would only watch one namespace. This practice has been discouraged, and most operators are written to be singletons on a cluster.

Tekton's operator uses the `TektonConfig` object for configuration, which is cluster-scoped. For us to depend on Tekton for installation, we should do the same.

See https://github.com/tektoncd/operator/blob/main/pkg/apis/operator/v1alpha1/tektonconfig_types.go#L33-L40

**Comment by otaviof** [comment]
Are we ready to merge this change?

**Comment by gabemontero** [comment]
good point @otaviof 

/lgtm

**Pull request merged**